src/vcpkg/configuration.cpp

```diff
@@ -387,6 +382,81 @@ namespace
         return ret;
     }
 
+    std::vector<LocalizedString> collect_package_pattern_warnings(const std::vector<RegistryConfig>& registries)
+    {
+        struct LocationAndRegistry
+        {
+            StringView location;
+            StringView registry;
+        };
+
+        // handle warnings from package pattern declarations
+        std::map<StringView, std::vector<LocationAndRegistry>> patterns;
+        for (auto&& reg : registries)
+        {
+            if (auto packages = reg.package_declarations.get())
+            {
+                for (auto&& pkg : *packages)
+                {
+                    auto it = patterns.find(pkg.pattern);
```

Comment:
Technically you could avoid a double lookup here by using `lower_bound` instead but I don't think it really matters.